perllib/FixMyStreet/App/Model/PhotoSet.pm

```diff
@@ -203,6 +215,12 @@ sub get_image_data {
     my $photo = $image->{data};
 
     my $size = $args{size};
+
+    if ($self->symlinkable && $size eq 'full') {
```

Comment:
If this also checked if `$image->{object}` was defined then we could remove the `$object` from the return of `S3::retrieve_photo`. (Otherwise, if I'm understanding it correctly, the `symlink` call in `Photo.pm` would crash/fail?)

Reply:
Yeah, and then that would also deal with your "storage backend to indicate whether it supports symlinks" too, good idea.